src/parser/jobs/drk/modules/ResourceSimulator.js

```diff
@@ -218,25 +213,41 @@ export default class Resources extends Module {
 	}
 
 	_onEvent(event) {
-		if (event.amount === 0) {
-			// Misses or attacks against invuln targets do not trigger resource gain
-			return
-		}
+		const _actionHit = (event.hitType !== HitType.MISS && event.hitType !== HitType.IMMUNE)
+
 		const abilityId = event.ability.guid
 		let actionBloodGain = 0
 		let actionMPGain = 0
+
 		if (RESOURCE_SPENDERS.hasOwnProperty(abilityId)) {
-			actionBloodGain += RESOURCE_SPENDERS[abilityId].mp
-			actionMPGain += RESOURCE_SPENDERS[abilityId].mp
+			if (RESOURCE_SPENDERS[abilityId].blood < 0 && this.combatants.selected.hasStatus(STATUSES.DELIRIUM.id)) {
+				// Blood spender under delirium - no change
+				actionBloodGain += 0
```

Comment:
I only just re-read this and realised you're `+= 0` which is basically a noop. If it significantly improves the readability to you I'm not gonna block on it, but it seems a little redundant.

Reply:
Noop was intentionally added for readability, since I can't return early out of the event there (using Bloodspiller under Delirium + Blood Weapon needs to trigger both that conditional plus the conditional for "using a GCD under Blood Weapon" further down, for a net of actionBloodGain = 10)